projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/lib/is-possible-to-extend-block.ts

```diff
@@ -24,3 +33,17 @@ export function isPossibleToExtendBlock( blockName: string ): boolean {
 
 	return true;
 }
+
+/**
+ * Hook that checks if a block can be extended with AI Assistant capabilities.
+ * @param {string} blockName - The block name.
+ * @return {boolean} Whether the block can be extended.
+ */
+export function useIsPossibleToExtendBlock( blockName: string ): boolean {
```

Comment:
I'd say the naming is a bit confusing. `isPossibleToExtendBlock` is already used to extend the block or not and I think it should be used only for that, as when this is called, the block is already extended.
This could be something like `useIsBlockExtensionEnabled`, so we can be in a state where extending the block is _possible_, but it is not _currently enabled_ for whatever reason, and this can change at any time.

Reply:
True true. Maybe the hook should make some reference to “module”, as it is what it’s really checking. Different from other extensions, this check is for “module dependency”. 
but overall agree it needs a name change and split it from isPossibleToExtendBlock